canonical/maas-ui
# feat(vault): Add status icons and warning message to controller list

## Done

- Add status icons to each region controller displaying the status of vault on them (if setup has started)
- Add a warning message to controller listing if there are unconfigured region controllers
- Add a warning message to controller listing if all region controllers are set up but secrets are not migrated

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /controllers
- Ensure no icons are displayed for vault status
- Ensure no warnings are displayed for vault

## Fixes

Fixes https://github.com/canonical/app-tribe/issues/1406

## Screenshots

### Unconfigured
![image](https://user-images.githubusercontent.com/35104482/196664130-e54fb59e-b02a-4842-9b65-b7e12a94a560.png)

### Partially configured
![image](https://user-images.githubusercontent.com/35104482/196663501-bae6473d-4b46-4620-8dd1-754f318e1e58.png)
![image](https://user-images.githubusercontent.com/35104482/196663547-1a426f2c-2b10-4e2e-b784-d168cabd5c94.png)
![image](https://user-images.githubusercontent.com/35104482/196663570-480756eb-bfba-42dc-b246-4a8bb9d60053.png)

### Configured but secrets not migrated
![image](https://user-images.githubusercontent.com/35104482/196663797-c17dae3b-31bb-4d8c-b43a-da84cf1eef3c.png)

### Vault fully configured
![image](https://user-images.githubusercontent.com/35104482/196663832-ca9e6936-f214-4a79-a604-a2d8c959cabd.png)
